contract_sale_generation/models/contract.py

```diff
@@ -0,0 +1,166 @@
+# © 2004-2010 OpenERP SA
+# © 2014 Angel Moya <angel.moya@domatix.com>
+# © 2015 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# © 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2017 Pesol (<http://pesol.es>)
+# Copyright 2017 Angel Moya <angel.moya@pesol.es>
+# Copyright 2018 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class ContractContract(models.Model):
+    _inherit = 'contract.contract'
+
+    sale_count = fields.Integer(compute="_compute_sale_count")
+
+    @api.multi
+    def _prepare_sale(self, date_ref):
+        self.ensure_one()
+        sale = self.env['sale.order'].new({
+            'partner_id': self.partner_id,
+            'date_order': date_ref.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
```

Comment:
I think is better to use fields.Date.to_string(date_ref), and also delete import